src/Kraken/Spot/API/Candle.jl

```diff
@@ -0,0 +1,127 @@
+module Candle
+
+export CandleQuery,
+    CandleData,
+    candle
+
+using Serde
+using Dates, NanoDates, TimeZones
+
+using CryptoAPIs.Kraken
+using CryptoAPIs.Kraken: Data
+using CryptoAPIs: Maybe, APIsRequest
+import CryptoAPIs: prepare_json!
+
+@enum TimeInterval m1 m5 m15 m30 h1 h4 d1 w1 d15
+
+Base.@kwdef struct CandleQuery <: KrakenPublicQuery
+    pair::String
+    interval::Maybe{TimeInterval} = nothing
+    since::Maybe{DateTime} = nothing
+end
+
+function Serde.ser_type(::Type{CandleQuery}, x::TimeInterval)::Int64
+    x == m1  && return 1
+    x == m5  && return 5
+    x == m15 && return 15
+    x == m30 && return 30
+    x == h1  && return 60
+    x == h4  && return 240
+    x == d1  && return 1440
+    x == w1  && return 10080
+    x == d15 && return 21600
+end
+
+function Serde.SerQuery.ser_type(::Type{CandleQuery}, x::DateTime)::Int64
+    return round(Int64, 1000 * datetime2unix(x))
+end
```

Comment:
Why is the method `set_type` additionally defined here for the `CandleQuery` type? If in Utils.jl already has an entry
```
function Serde.SerQuery.ser_type(::Type{<:KrakenCommonQuery}, dt::D)::Int64 where {D<:DateTime}
    return round(Int64, 1000 * datetime2unix(dt))
end
```